connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -309,6 +310,28 @@ class SparkConnectPlanner(session: SparkSession) {
     UnresolvedHint(rel.getName, params, transformRelation(rel.getInput))
   }
 
+  private def transformMelt(rel: proto.Melt): LogicalPlan = {
+    val ids = rel.getIdsList.asScala.toArray.map { expr =>
+      Column(transformExpression(expr))
+    }
+
+    if (rel.getValuesList.isEmpty) {
+      Dataset
+        .ofRows(session, transformRelation(rel.getInput))
+        .melt(ids, rel.getVariableColumnName, rel.getValueColumnName)
```

Comment:
What would happen if you pass a empty values? If that works then we can remove this if else?

Reply:
It will throws
```
org.apache.spark.sql.AnalysisException: [UNPIVOT_REQUIRES_VALUE_COLUMNS] At least one value column needs to be specified for UNPIVOT, all columns specified as ids;
'Unpivot ArraySeq(id#25), ArraySeq(), variable, [value]
+- LocalRelation <empty>, [id#25, name#26]
```